service/history/task/task.go

```diff
@@ -262,18 +266,23 @@ func (t *taskImpl) HandleErr(err error) (retErr error) {
 		return nil
 	}
 
-	if t.GetAttempt() > stickyTaskMaxRetryCount && common.IsStickyTaskConditionError(err) {
+	attempt := t.GetAttempt()
```

Comment:
who is incrementing attempt? shouldn't it be done in this Execute function?

Reply:
it's done by a defer function in this HandleErr method.